File: Training/1.2 Beginning/1.2.4-beads.py
'''
ID:   andy1li
LANG: PYTHON3
TASK: beads
'''

from collections import deque

def collect(s, color = ''):
    for i, x in enumerate(s):
        if x == 'w': continue
        if not color: color = x
        elif x != color: break
    return i

def solve():
    n = int(input())
    beads = deque(input().strip())
    
    ans = 0
    for i in range(n):
        collected = collect(beads) + collect(reversed(beads))
        ans = max(ans, collected)
        beads.rotate()
    print_line(min(n, ans))

#------------------------------------------------------------------------------#

def print_line(line): print(line); f_out.write( f'{line}\n' )

task = 'beads'
with open(task+'.in') as f_in, open(task+'.out', 'w') as f_out:
    input = f_in.readline
    solve()

File: Training/1.2 Warm up/1.2.4-beads.cpp
/*
ID:   andy1li
LANG: C++
TASK: beads 
*/

#include <bits/stdc++.h>
using namespace std;

#define all(xs) xs.begin(), xs.end()
#define FOR(stop) for (size_t i=0; i<stop; ++i)

/*----------------------------------------------------------------------------*/

int collect(string s, char color = 'w') {
    FOR(s.size()) {
        char x = s[i];
        if (x == 'w') continue;
        if (color == 'w') color = x;
        else if (x != color) return i;
    }
    return s.size();
}

void solve() {
    int n; cin >> n;
    string beads; cin >> beads;

    int ans = 0;
    FOR(n) {
        string rbeads = beads; reverse(all(rbeads));
        ans = max(ans, collect(beads) + collect(rbeads));

        rotate(begin(beads), begin(beads)+1, end(beads));
    }
    cout << min(n, ans) << endl;
}

/*----------------------------------------------------------------------------*/

int main() {
    freopen("beads.in",  "r", stdin);
    freopen("beads.out", "w", stdout);
    solve();
}